lib/companies/job_checker.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Companies.JobChecker do
+  @moduledoc """
+  The module to handle automated PendingChange creation for the no longer present jobs.
+  """
+
+  alias Companies.{Jobs, PendingChanges}
+
+  def check_job(%{url: url} = job) do
+    case HTTPoison.head(url) do
+      {:ok, %{status_code: 404}} ->
+        user_id = Application.get_env(:companies, :jobs_url_checker)[:user_id]
```

Comment:
`Application.get_env(:companies, :jobs_url_checker)[:user_id]` resolves to `System.get_env("JOB_CHECKER_USER_ID")` in a production environment, and `nil` in other environments. Would it be simpler to read from the environment variable here and omit setting `user_id` in the configuration file?
```ex
user_id = System.get_env("JOB_CHECKER_USER_ID")
```
That has a slight potential benefit of allowing you to change the value without restarting the app, although I'm not sure if that will make a practical difference. It also simplifies the application logic a little bit. A minor downside is that a change like this splits the configuration up between the configuration file and the application code. It may also be detrimental to use the `JOB_CHECKER_USER_ID` environment variable in non-production environments if that value could be non-empty, and you wanted to ensure `user_id` was `nil` in those environments.

Just some food for thought :smile: